leankyr/Warmup_Problems/Utopian_Tree/Utopian_Tree.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3.5
+from __future__ import print_function
+
+import math
+import os
+import random
+import re
+import sys
+import logging
+#import ipdb
+
+logging.basicConfig(level=logging.WARN, format='%(levelname)s- %(message)s')
+
+
+
+def read_file(filename):
+    # Define filename
+    #filename = "testfile.txt"
+    # Define file_object with the above file_name
+    file_obj = open(filename, "r")
+    # DEBUG
+    # store size and print it to make sure
+    n = file_obj.readline()
+    if(n >= 10 or n < 0):
+        return -1
+    # print size
+    # print("(Comes From Function)n is: ", n)
+    # define empty matrix 
+    array = []
+    for i in range(0,int(n)):
+        # INFO Converted to int in order to dish the new line character that
+        # comes with readline
+        # HashTag TsompanDev
+        if(int(file_obj.readline()) > 60 ):
+            return -1
+
+        array.append(int(file_obj.readline()))
+
+    #print final matrix for debug 
+    #for i in range(0,len(array)):
+    #    print (array[i],end="")
+    return array
+
+
+
+def utopianTree(n):
+    if(n==0):
+        return 1
+    # QUESTION:
```

Comment:
I think you need [reduce](https://www.python-course.eu/python3_lambda.php) for that